examples/Microsoft.Spark.CSharp.Examples/Sql/Datasource.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Spark.Sql;
+
+
+namespace Microsoft.Spark.Examples.Sql
+{
+    /// <summary>
+    /// A simple example demonstrating Spark SQL data sources
+    ///
+    ///  A copy of spark/examples/src/main/python/sql/datasource.py
```

Comment:
Can we do something similar to datasource.py?

`Run()` will call `RunBasicDataSourceExample()`, `RunParquetExample()`, `RunJsonDatasetExample()`.
Note that `parquet_schema_merging_example` and `json_dataset_example` cannot be supported.

But great start! Thanks for your contribution!

Reply:
yep will sort! :)

Reply:
i've changed it so it matches the python version and added the jdbc example - hopefully this now matches the style guidelines as well :)